### ISSUE TITLE
What is performance and convert back to anisotropic?

### ISSUE THREAD

**Issue by John1231983** [Yt-trium/DeepSeg3D#1]
Thanks for great work. I have two questions:
1. What is DSC performance?
2. If you convert the image to isotropic, but the grouth-truth for evaluation must be anisotropic, How to convert back to anisotropic for computing DSC? Does converting isotropic helps improve performance? 

Thanks

**Comment by Yt-trium** [comment]
Hi,

1. For the moment, the best results were in the range of 0.75 ~ 0.80 on the training dataset and 0.55 ~ 0.65 on the validation and testing dataset. It really depends on the model, patchs size, and data augmentation we use.

2. In our case, we convert the images and the ground truth, so we don't have such problem.

The main ideas behind the isotropic images are the following.
First, allow the network to predict correctly on other images (other datasets) having the same proportion and dimensions.
Secondly, increase the data by allowing patches to rotate at 90 degrees on all axes
Third, teach the models the topological structure of the images

You can convert back to anisotropic anytime you want using the same process we use to make them isotropic.

The conversion doesn't make the model better on our single dataset (Bullitt) but it helps to adapt on every type of images.

**Comment by John1231983** [comment]
Good points. Actually, I am working same problem for MRBrainS13 dataset. It is the anisotropic resolution likes 0.3x0.3x1mm. If I convert the resolution to 1mmx1mmx1mm and apply CNN, it worked fine but the ground-truth for testing are keeping on the server and it is 0.3x0.3x1, thus I have to convert back to the original resolution for evaluation. But the converting back often has errors in pixel level, and it also affects to the final result. Without converting to isotropic, we can ignore the error in the converting back process.

**Comment by Yt-trium** [comment]
If you only want to make the best score on a specific dataset, you better don't make the image isotropic.
The conversion will of course create some "errors" due to the interpolation which is never exact, and in your case you want to have some "overfitting".

**Comment by John1231983** [comment]
Thanks. I think your dsc can improve by using ensembling networks, better feature extraction, dice loss...I do not think augmentation can help. I often use augmentation but it hurts my performance. Does it improve your DSC?

**Comment by Yt-trium** [comment]
Yes, it really help us, mainly because we only have 38 images for the training.

I will check about ensembling networks, I don't know it. What do you mean by "better feature extraction" ?

**Comment by John1231983** [comment]
It means using resnetxt, inception....ensembling means combined different results from different models in testing. I just have 5 subject for training. What is techniques for your augmentation? I am using flip left right and up down. I did not using scaling becausevthe error in the bilinear iterpolation

**Comment by Yt-trium** [comment]
Ok, so I didn't know the word, but we are already making ensembling networks (3 different models are trained and used for predictions).

Patchs are extracted from random images at a random position and randomly rotated on 3 axis.
Yes I love things to be random.

**Comment by John1231983** [comment]
Hi again, I do not want to open new issue because this is the question related to performance, not a bug. Could you tell me the performance do you get using spatial dropout in comparison with dropout. My performances are same when using two dropouts

**Comment by Yt-trium** [comment]
I have not noticed a difference yet, but the experiments are just beginning.
